code/modules/atmospherics/machinery/components/fusion/hypertorus.dm

```diff
@@ -0,0 +1,1376 @@
+		. = FALSE
+
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/proc/activate(mob/living/user)
+	if(active)
+		to_chat(user, "<span class='notice'>You already activated the machine.</span>")
+		return
+	to_chat(user, "<span class='notice'>You link all parts toghether.</span>")
+	active = TRUE
+	update_icon()
+	linked_interface.active = TRUE
+	linked_interface.update_icon()
+	linked_input.active = TRUE
+	linked_input.update_icon()
+	linked_output.active = TRUE
+	linked_output.update_icon()
+	linked_moderator.active = TRUE
+	linked_moderator.update_icon()
+	for(var/obj/machinery/hypertorus/corner/corner in corners)
+		corner.active = TRUE
+		corner.update_icon()
+	soundloop = new(list(src), TRUE)
+	soundloop.volume = 5
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/proc/deactivate()
+	if(!active)
+		return
+	active = FALSE
+	update_icon()
+	if(linked_interface)
+		linked_interface.active = FALSE
+		linked_interface.update_icon()
+	if(linked_input)
+		linked_input.active = FALSE
+		linked_input.update_icon()
+	if(linked_output)
+		linked_output.active = FALSE
+		linked_output.update_icon()
+	if(linked_moderator)
+		linked_moderator.active = FALSE
+		linked_moderator.update_icon()
+	if(corners.len)
+		for(var/obj/machinery/hypertorus/corner/corner in corners)
+			corner.active = FALSE
+			corner.update_icon()
+	QDEL_NULL(soundloop)
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/proc/check_fuel()
+	if(internal_fusion.gases[/datum/gas/tritium][MOLES] > FUSION_MOLE_THRESHOLD && internal_fusion.gases[/datum/gas/hydrogen][MOLES] > FUSION_MOLE_THRESHOLD)
+		return TRUE
+	return FALSE
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/proc/check_power_use()
+	if(machine_stat & (NOPOWER|BROKEN))
+		return FALSE
+	if(use_power == ACTIVE_POWER_USE)
+		active_power_usage = ((power_level + 1) * MIN_POWER_USAGE) //Max around 350 KW
+	return TRUE
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/proc/get_status()
+	var/integrity = get_integrity()
+	if(integrity < HYPERTORUS_MELTING_PERCENT)
+		return HYPERTORUS_MELTING
+
+	if(integrity < HYPERTORUS_EMERGENCY_PERCENT)
+		return HYPERTORUS_EMERGENCY
+
+	if(integrity < HYPERTORUS_DANGER_PERCENT)
+		return HYPERTORUS_DANGER
+
+	if(integrity < HYPERTORUS_WARNING_PERCENT)
+		return HYPERTORUS_WARNING
+
+	if(power_level > 0)
+		return HYPERTORUS_NOMINAL
+	return HYPERTORUS_INACTIVE
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/proc/alarm()
+	switch(get_status())
+		if(HYPERTORUS_MELTING)
+			playsound(src, 'sound/misc/bloblarm.ogg', 100, FALSE, 40, 30, falloff_distance = 10)
+		if(HYPERTORUS_EMERGENCY)
+			playsound(src, 'sound/machines/engine_alert1.ogg', 100, FALSE, 30, 30, falloff_distance = 10)
+		if(HYPERTORUS_DANGER)
+			playsound(src, 'sound/machines/engine_alert2.ogg', 100, FALSE, 30, 30, falloff_distance = 10)
+		if(HYPERTORUS_WARNING)
+			playsound(src, 'sound/machines/terminal_alert.ogg', 75)
```

Comment:
If this is the only use of get_status, why are there defines that go unused?

Reply:
still need to cleanup, will to it later